server/resource_server/service/service.go

```diff
@@ -4,25 +4,28 @@ import (
 	"github.com/go-playground/validator/v10"
 	"globe-and-citizen/layer8/server/resource_server/dto"
 	"globe-and-citizen/layer8/server/resource_server/emails/verification"
-	interfaces "globe-and-citizen/layer8/server/resource_server/interfaces"
+	"globe-and-citizen/layer8/server/resource_server/emails/verification/zk"
+	"globe-and-citizen/layer8/server/resource_server/interfaces"
 	"globe-and-citizen/layer8/server/resource_server/models"
 	"globe-and-citizen/layer8/server/resource_server/utils"
 	"time"
 )
 
 type service struct {
-	repository    interfaces.IRepository
-	emailVerifier *verification.EmailVerifier
+	repository       interfaces.IRepository
+	emailVerifier    *verification.EmailVerifier
+	zkProofProcessor *zk.ProofProcessor
```

Comment:
3 months ago when you first saw the opportunity with Globe & Citizen, did you have any idea you'd be writing this line of code, "zkProofProcessor *zk.ProofProcessor"?

Reply:
Of course, not :) I didn't even hear what zk-snarks were about!